### PR TITLE
Add recipe for micro_sam

### PR DESCRIPTION
**Pull request by constantinpape**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by constantinpape** [comment]
It seems like there is no OpenGL in the linux build machine available, so the tests fail there. (See the full error log below).
I only found [this](https://conda-forge.org/docs/maintainer/knowledge_base.html#libgl) in the Knowledgebase, but it refers to packages that link against libGL, which is not the case here. This package depends on `napari` (https://napari.org/stable/), but it is a noarch python package that does not link to libGL (or to any other library).

@conda-forge/help-python any hint for how to fix this OpenGL related issue?
Thanks! 

```
======================================================================
ERROR: test_compute_iou (test_util.TestUtil.test_compute_iou)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/conda/staged-recipes/build_artifacts/micro_sam_1689670813821/test_tmp/test/test_util.py", line 8, in test_compute_iou
    from micro_sam.util import compute_iou
  File "/home/conda/staged-recipes/build_artifacts/micro_sam_1689670813821/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.11/site-packages/micro_sam/util.py", line 22, in <module>
    from napari.utils import progress as tqdm
  File "/home/conda/staged-recipes/build_artifacts/micro_sam_1689670813821/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.11/site-packages/napari/utils/__init__.py", line 2, in <module>
    from napari.utils.colormaps import Colormap
  File "/home/conda/staged-recipes/build_artifacts/micro_sam_1689670813821/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.11/site-packages/napari/utils/colormaps/__init__.py", line 2, in <module>
    from napari.utils.colormaps.colormap import Colormap
  File "/home/conda/staged-recipes/build_artifacts/micro_sam_1689670813821/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.11/site-packages/napari/utils/colormaps/colormap.py", line 7, in <module>
    from napari.utils.color import ColorArray
  File "/home/conda/staged-recipes/build_artifacts/micro_sam_1689670813821/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.11/site-packages/napari/utils/color.py", line 7, in <module>
    from napari.utils.colormaps.standardize_color import transform_color
  File "/home/conda/staged-recipes/build_artifacts/micro_sam_1689670813821/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.11/site-packages/napari/utils/colormaps/standardize_color.py", line 27, in <module>
    from vispy.color import ColorArray, get_color_dict, get_color_names
  File "/home/conda/staged-recipes/build_artifacts/micro_sam_1689670813821/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.11/site-packages/vispy/color/__init__.py", line 12, in <module>
    from .colormap import (Colormap, BaseColormap,  # noqa
  File "/home/conda/staged-recipes/build_artifacts/micro_sam_1689670813821/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.11/site-packages/vispy/color/colormap.py", line 14, in <module>
    import vispy.gloo
  File "/home/conda/staged-recipes/build_artifacts/micro_sam_1689670813821/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.11/site-packages/vispy/gloo/__init__.py", line 47, in <module>
    from . import gl  # noqa
    ^^^^^^^^^^^^^^^^
  File "/home/conda/staged-recipes/build_artifacts/micro_sam_1689670813821/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.11/site-packages/vispy/gloo/gl/__init__.py", line 230, in <module>
    from . import es2 as default_backend  # noqa
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/conda/staged-recipes/build_artifacts/micro_sam_1689670813821/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/python3.11/site-packages/vispy/gloo/gl/es2.py", line 48, in <module>
    raise OSError('GL ES 2.0 library not found')
OSError: GL ES 2.0 library not found
```

**Comment by ocefpaf** [comment]
Take a look at https://conda-forge.org/docs/maintainer/maintainer_faq.html#mfaq-libgl-so-1, it may help here.

**Comment by constantinpape** [comment]
@conda-forge-admin please rerender

**Comment by constantinpape** [comment]
@ocefpaf I followed the steps outlined in https://conda-forge.org/docs/maintainer/maintainer_faq.html#mfaq-libgl-so-1 to add lib-mesa-gl to the yum requirements. However, for this to take effect the recipe needs to be rerendered.
But I think `@conda-forge-admin please rerender` does not work here since it's a PR onto `staged-recipes` and not onto a feedstock.
How do I rerender this PR?

**Comment by constantinpape** [comment]
Adding the other yum requirements didn't help.
I have cross-posted this on image.sc now: https://forum.image.sc/t/conda-forge-recipe-with-napari-dependency/83903
Maybe someone there with napari experience could help.

**Comment by jni** [comment]
I hope it's ok to ping @jaimergp and @goanpeca here... 🙏

**Comment by constantinpape** [comment]
Hi @ocefpaf ,
I decided to remove napari from the dependencies to fix this issue. This is better anyways, because `napari` is only an optional dependency anyways, it's needed for the interactive software included here, but not for the library component.
I tried to express the optional dependency in the conda-recipe but couldn't really find a way to do this (the closest match I found was this issue: https://github.com/conda/conda/issues/1696. But it does not state how exactly optional dependencies are handled despite the issue claiming that this was implemented in conda v4.4. I also couldn't find any documentation on this.)

In any case, I think this is ready for review now, with the following comments from my side:
- The build error on windows can be ignored since this is a `noarch` package. (And the error with installing the requested packages can be solved in practice by adding the `pytorch` channel, so it will still work for windows users).
- As mentioned above I am not sure if there is any way of expressing the optional dependency on `napari`

**Comment by ocefpaf** [comment]
I don't think that issue was closed b/c optional dependencies were implemented. Conda cannot handle that in the same fashion as Python b/c conda is not a Python package manager. What one can do is to use multiple outputs to represent the optional variants of a package with different names. See https://github.com/conda-forge/apache-beam-feedstock/blob/main/recipe/meta.yaml as an example.

With that said, sometimes multiple outputs packages are too complex and not worth the trouble. Let's merge this as-is and you can decide what to do in the feedstock.

**Comment by constantinpape** [comment]
Thanks for merging and for providing the context about the packaging options, @ocefpaf! 

And I don't think the multiple outputs are necessary here, I will just note that napari needs to be installed as an additional dependency to use the gui functionality in the installation instructions.

**Comment by constantinpape** [comment]
Did anything go wrong with creating a feedstock? I didn't get any notifications about this yet, and while here is https://github.com/conda-forge/micro_sam-feedstock, it seems to be completely empty.

**Comment by constantinpape** [comment]
Note to self: should use `run_constrained` for specifying the valid napari version: https://github.com/conda/conda/issues/1696

**Comment by constantinpape** [comment]
Hi @ocefpaf and @conda-forge/help-python ,
It looks like something went wrong when merging this PR. I did not get any notifications from the feedstock. I could find it in the `conda-forge` repo, but it is empty: https://github.com/conda-forge/micro_sam-feedstock
Any idea what went wrong and how to fix it?

Thanks a lot for the help!

File: recipes/yum_requirements.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+mesa-libGL-devel
```
Comment (beckermr) [inline recipes/yum_requirements.txt:1]: Looks like this file not being in the recipe dir broke staged recipes. We should add something to smithy to lint for this. cc @ocefpaf

**Comment by beckermr** [comment]
I fixed it on main and things should start working again.

**Pull request merged**